README.ideal.md

```diff
@@ -0,0 +1,131 @@
+Ni-Mh (called “nickel-nix” or “Nixel” at the moment, but I really like Ni-Mh) aims at being your main entrypoint for managaing all these different tools, so that you can get:
+
+1. A unified configuration framework for all of these
+2. A powerful and ergonomic language to allow you to easily abstract over these configurations, with discoverability and early error reporting
+3. A way to get these tools to communicate with each others, so that your custom scripts can easily leverage your package manager to get the dependencies they need, your CI setup can understand your build system and parallelise efficiently, and so on.
+
+## Getting started
+
+To start using `nickel-nix`, you need `Nix` to be installed and configured.
+If it isn't already the case, you can get it with:
+
+```console
+$ curl -L https://nixos.org/nix/install | bash
+# We also need a couple of experimental Nix features
+$ mkdir -p ~/.config/nix
+$ echo 'experimental-features = nix-command flakes' >> ~/.config/nix/nix.conf
+```
+
+Then bootstrap a project with:
+
+```console
+$ nix flake init -t github:nickel-lang/nickel-nix
+# Edit the project file to fit your needs
+$ $EDITOR project.ncl
+# Enter the environment
+$ . .nickel-nix/activate
+```
+
+## Managing your environment with Nixel
+
+### Dependencies
+
+Nixel can be used to declare the dependencies for your project.
+These can then be instantiated using [Nix](https://nixos.org/nix).
+
+> TODO: Document
+
+### CI configuration
+
+You can use Nixel to generate a CI configuration.
+This is extremely handy if you want to test some non-trivial matrix of platforms/configuration.
+
+Only Github actions is supported at the moment, but we plan to add more options soon.
+
+> TODO: Document
+
+### Development services
+
+Nixel can also be used for some simple service management.
+In combination with the Nix integration, this allows you to quickly spawn any service that you might depend on.
+
+Once the services have been defined in your `project.ncl` file, you can run `nixel up` to start them.
+
+> TODO: Document
+
+### Deployment
+
+If you want to push things yet a bit further, Nixel can also interoperate with [tf-ncl](https://github.com/tweag/tf-ncl) to allow you to manage the deployment infrastructure for your project.
+
+> TODO: Document
+
+## Structure of the `project.ncl` file
+
+`project.ncl` is the main entrypoint for `nickel-nix`.
+It describes the packages that should be included in your development environment, as well as the template for any file you'd like to generate:
+
+```nickel
+let Nixel = import ".nickel-nix/lock.ncl" in
+{
+  nix_inputs.spec = {
+    pandoc = {},
+    make = {},
+    prettier = {},
+    redis = {},
```

Comment:
Can there be something in the `{}`? If so, maybe an example of such would be worth it?

Also, why is it called `nix_inputs.spec` and not just `nix_inputs`?

Reply:
> Can there be something in the {}? If so, maybe an example of such would be worth it?

Added that

> Also, why is it called nix_inputs.spec and not just nix_inputs?

That's a consequence of the evaluation model where `nix_inputs.spec` (currently `inputs_spec`) is the specification of the inputs, and their actual value is injected into `inputs`. But maybe we can do some smart merging wizardry to only define parts of `inputs.foo` and inject the rest of the value. Not sure yet.

Reply:
We could also get totally rid of this `nix_inputs` thing by changing the evaluation model (as described in https://github.com/nickel-lang/nickel-nix/issues/56). There's a tradeoff waiting to be investigated here.

Reply:
Writing this more in depth, I actually find this `nix_inputs` field quite annoying. Either it's at toplevel, and it's OK but not composable-enough (in particular setting `shells = Nixel.shells.Rust` can't touch it), or it's per-section in the config file, and it can quickly get quite boilerplaty

Reply:
Very well, let's change this ideal README to feature a better solution, then.

Reply:
I've tried it another way. Let me know how you like it

Reply:
Yeah, still not wonderful, I agree. I'm taker of any suggestion for something nicer